### ISSUE TITLE
twisted.internet.iocpreactor.iocpsupport.get_accept_addrs should work with AF_INET6 sockets too

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5109]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5109|
|Type|enhancement|
|Created|2011-05-15 13:45:19Z|
|Branch|https://github.com/twisted/twisted/tree/iocp-af_inet6-5109|

Currently get_accept_addrs only supports AF_INET sockets.  This prevents iocpreactor from completely supporting IPv6, since it won't be able to report the addresses of AF_INET6 sockets.

<details><summary>Searchable metadata</summary>

```
trac-id__5109 5109
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_iocp_af_inet6_5109 branches/iocp-af_inet6-5109
branch_author__PenguinOfDoom__exarkun PenguinOfDoom, exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__windows_ipv6 windows ipv6
time__1305467119000000 1305467119000000
changetime__1307141747000000 1307141747000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [31867]) Branching to 'iocp-af_inet6-5109'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/iocp-af_inet6-5109), please review.

**Comment by twisted-trac** [comment]
|[<img alt="djfroofy's avatar" src="https://avatars.githubusercontent.com/u/89760?s=50" width="50" height="50">](https://github.com/djfroofy)<a name="note_3"></a>|@djfroofy set owner to @djfroofy<br>@djfroofy set status to `assigned`|
|-|-|

+1. There are tests failures but same intermittent ones as trunk and known ([#3404](https://github.com/twisted/twisted/issues/3404)).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [32028]) Merge iocp-af_inet6-5109

Author: PenguinOfDoom, exarkun
Reviewer: djfroofy
Fixes: [#5109](https://github.com/twisted/twisted/issues/5109)

Add AF_INET6 support to the internal IOCP reactor address helper so that
when it encounters an AF_INET6 socket it can properly report its address.
